modules/h12mediaBidAdapter.js

```diff
@@ -40,63 +48,67 @@ export const spec = {
         adunitSize: bidRequest.mediaTypes.banner.sizes || [],
         coords,
         ishidden,
+        pubsubid,
+        pubcontainerid,
       };
-    });
 
-    return {
-      method: 'POST',
-      url: requestUrl,
-      options: {withCredentials: false},
-      data: {
-        gdpr: utils.deepAccess(bidderRequest, 'gdprConsent.gdprApplies') ? Boolean(bidderRequest.gdprConsent.gdprApplies & 1) : false,
-        gdpr_cs: utils.deepAccess(bidderRequest, 'gdprConsent.gdprApplies') ? bidderRequest.gdprConsent.consentString : '',
-        topLevelUrl: window.top.location.href,
-        refererUrl: bidderRequest.refererInfo ? bidderRequest.refererInfo.referer : '',
-        isiframe,
-        version: '$prebid.version$',
-        visitorInfo: {
-          localTime: getLocalDateFormatted(),
-          dayOfWeek: new Date().getDay(),
-          screenWidth: screenSize[0],
-          screenHeight: screenSize[1],
-          docWidth: docSize[0],
-          docHeight: docSize[1],
-          scrollbarx: window.scrollX,
-          scrollbary: window.scrollY,
+      return {
+        method: 'POST',
+        url: requestUrl,
+        options: {withCredentials: true},
+        data: {
+          gdpr: !!utils.deepAccess(bidderRequest, 'gdprConsent.gdprApplies', false),
+          gdpr_cs: utils.deepAccess(bidderRequest, 'gdprConsent.consentString', ''),
+          usp: !!utils.deepAccess(bidderRequest, 'uspConsent', false),
+          usp_cs: utils.deepAccess(bidderRequest, 'uspConsent', ''),
+          topLevelUrl: utils.deepAccess(bidderRequest, 'refererInfo.referer', ''),
+          refererUrl: window.top.document.referrer || window.document.referrer,
```

Comment:
you should guard `window.top` access with a `try-catch`, same goes for other places where you use it.

Reply:
Fixed, `try-catch` with `window.top` added to all places where it was missing